### PR TITLE
Provision dynamic inventory

### PR DESCRIPTION
**Pull request by danlamanna**
This depends on #171 and #175.

This is a rough draft/WIP - @kotfic we can talk about this tomorrow.

### CODE REVIEW

**Comment by cjh1** [comment]
@kotfic @danlamanna What is the state of the PR? What do we need todo to move it alone?

File: girder/cumulus/server/models/cluster.py [hunk 1]
```diff
@@ -207,9 +207,25 @@ def create_newt(self, user, name, config):
...
     def add_log_record(self, user, id, record):
+        def mongo_safe_value(value):
+            new_value = {}
+
+            if not isinstance(value, dict):
+                return value
+            else:
+                for (k, v) in value.iteritems():
+                    if '.' in str(k) or '$' in str(k):
+                        k = k.replace('.', '_').replace('$', '_')
```
Comment (cjh1) [inline girder/cumulus/server/models/cluster.py]: Could we replace with the unicode chars, so we don't loss information?

**Comment by kotfic** [comment]
@cjh1 This looks good to me,  but i am not really qualified to test the SGE part - you can follow the [README.md](https://github.com/Kitware/cumulus/blob/feature-cumulus-provision-dynamic-inventory/cumulus/ansible/README.md) to get a cluster up and running.

**Comment by cjh1** [comment]
@kotfic Can we update the ansible integration test so it does the provisioning step as well?

File: girder/cumulus/server/models/cluster.py (hunk 1 above)
Comment (cjh1) [inline girder/cumulus/server/models/cluster.py]: @danlamanna Not sure if you missed this comment?

File: cumulus/ansible/tasks/cluster.py [hunk 2]
```diff
@@ -90,3 +75,46 @@ def run_ansible(cluster, profile, secret_key, extra_vars,
...
+    playbook = get_playbook_path(playbook)
+    playbook_variables = get_playbook_variables(cluster, profile, extra_vars)
+
+    env = os.environ.copy()
+    env.update({"AWS_ACCESS_KEY_ID": profile['accessKeyId'],
+                "AWS_SECRET_ACCESS_KEY": secret_key,
+                "GIRDER_TOKEN": girder_token,
+                "LOG_WRITE_URL": log_write_url,
+                "CLUSTER_ID": cluster["_id"]})
+
```
Comment (cjh1) [inline cumulus/ansible/tasks/cluster.py]: Can we use single quotes here please and everywhere else

Comment (kotfic) [inline cumulus/ansible/tasks/cluster.py]: Sorry,  this got fixed,  but had to be un-fixed to resolve conflicts when merging in master (with the gpl-refactor branch). I thought we re-fixed it,  but apparently not :confounded:

File: girder/cumulus/server/models/cluster.py [hunk 3]
```diff
@@ -215,7 +215,7 @@ def mongo_safe_value(value):
             else:
                 for (k, v) in value.iteritems():
                     if '.' in str(k) or '$' in str(k):
-                        k = k.replace('.', '_').replace('$', '_')
+                        k = k.replace('.', u'_').replace('$', u'_')
```
Comment (cjh1) [inline girder/cumulus/server/models/cluster.py]: Sorry I didn't make myself clear. What I meant, was the following:
k = k.replace('.', '\\u002e').replace('$', '\\u0024')

**Pull request merged**